src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateFilter.java

```diff
@@ -33,9 +33,9 @@
     public static List<PodTemplate> applyAll(@Nonnull KubernetesCloud cloud, @Nonnull List<PodTemplate> podTemplates, @CheckForNull Label label) {
         List<PodTemplate> result = new ArrayList<>();
         for (PodTemplate t : podTemplates) {
-            PodTemplate output = null;
```

Comment:
Note that this changes behavior in the case that there are _no_ filters. Previously, this function would return an empty list, which makes little sense; now it will return (a copy of) `podTemplates` unchanged, as you would expect. I suppose that bug was not previously noticed because `PodTemplateLabelFilter` is always registered.